### ISSUE TITLE
POSIX-TZ and last transition time

### ISSUE THREAD

**Issue by yorickhardy** [daurnimator/luatz#17]
I tried the following code fragment:
```
local luatz = require("luatz")

local sydney_ts = luatz.time_in("Australia/Sydney")
sydney = luatz.timetable.new_from_timestamp(sydney_ts)

print(sydney)
print(os.date("!%c", sydney_ts))
os.execute("env TZ=Australia/Sydney date")

```
and the output:
```
2023-03-15T15:58:08.000
Wed Mar 15 15:58:08 2023
Wed Mar 15 16:58:08 AEDT 2023

```
differs by an hour. Looking through the code, it seems like the TZ entry is not used "in handling instants after the last transition time stored in the file" as per the documentation for the Version 2 tzfile format:

[Linux tzfile.5]( https://linux.die.net/man/5/tzfile)
[NetBSD tzfile.5](https://man.netbsd.org/tzfile.5)

I am not sure that I have read everything correctly though.

**Comment by daurnimator** [comment]
Works for me:

```lua
luatz = require("luatz")

sydney_ts = luatz.time_in("Australia/Sydney")
sydney = luatz.timetable.new_from_timestamp(sydney_ts)

print(sydney)
print(os.date("!%c", math.floor(sydney_ts)))
os.execute("env TZ=Australia/Sydney date")
```

```
$ lua 17.lua
2023-03-21T13:19:30.597
Tue Mar 21 13:19:30 2023
Tue 21 Mar 2023 13:19:30 AEDT
```

Are you able to attach your tzfile?

**Comment by yorickhardy** [comment]
Thanks! Maybe I need to update by tz files ...

I have attached the relevant file below.

On 2023-03-20, daurnimator wrote:
> Works for me:
> 
> ```lua
> luatz = require("luatz")
> 
> sydney_ts = luatz.time_in("Australia/Sydney")
> sydney = luatz.timetable.new_from_timestamp(sydney_ts)
> 
> print(sydney)
> print(os.date("!%c", math.floor(sydney_ts)))
> os.execute("env TZ=Australia/Sydney date")
> ```
> 
> ```
> $ lua 17.lua
> 2023-03-21T13:19:30.597
> Tue Mar 21 13:19:30 2023
> Tue 21 Mar 2023 13:19:30 AEDT
> ```
> 
> Are you able to attach your tzfile?

-- 
Kind regards,

Yorick Hardy

**Comment by yorickhardy** [comment]
Oops, github does not seem to like the attachment. Let me try again (in a zip file).
[sydney.zip](https://github.com/daurnimator/luatz/files/11026891/sydney.zip)
